File: src/main/kotlin/fr/jhelp/kotlinLightSamples/WhenSample.kt
package fr.jhelp.kotlinLightSamples

fun whenWithParameter(parameter: Int)
{
    when (parameter)
    {
        0 -> println("ZERO")
        5 -> println("FIVE")
        25 -> println("Twenty five")
        42 -> println("The answer")
        73 -> println("Magic Number")
        666 ->
        {
            println("Don't play with it : ")
            println("   It is the Beast !!!")
        }
        else -> println("Nothing to say for $parameter")
    }
}

fun whenWithoutParameter(integer: Int, string: String)
{
    when
    {
        integer == 5                       -> println("FIVE")
        integer < 5 && string.isNotEmpty() -> println("$integer => $string")
        integer > 5 && string.isEmpty()    ->
        {
            println("Best car ever : DeLoreen 1985")
        }
        else                               ->
        {
            println("I have a machine gun now! Ho! Ho! Ho!")
        }
    }
}

File: src/main/swift/Tests/swiftTests/src/jhelp/kotlinLightSamples/WhenSample.swift

public func whenWithParameter(_ parameter: Int)
{
    switch parameter
    {
        case 0  :  print("ZERO")
        case 5  :  print("FIVE")
        case 25  :  print("Twenty five")
        case 42  :  print("The answer")
        case 73  :  print("Magic Number")
        case 666  : 
        do {
            print("Don't play with \(0) : ")
            print("   It is the Beast !!")
        }
        default  :  print("Nothing to say for \(parameter)")
    }
}

public func whenWithoutParameter(_ integer: Int, _ string: String)
{
    
        if integer == 5 { print("FIVE")
       } else if integer < 5 && string.isNotEmpty() { print("\(integer) => \(string)")
       } else if integer > 5 && string.isEmpty() {
            print("Best car ever : DeLoreen 1985")
        }
        else {
            print("I have a machine gun now! Ho! Ho! Ho!")
        }
    
}
